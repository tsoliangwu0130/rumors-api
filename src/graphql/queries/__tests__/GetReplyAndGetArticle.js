import gql from 'util/GraphQL';
import { loadFixtures, unloadFixtures } from 'util/fixtures';
import fixtures from '../__fixtures__/GetReplyAndArticle';

describe('GetReplyAndGetArticle', () => {
  beforeAll(() => loadFixtures(fixtures));

  describe('GetArticle', () => {
    it('should get the specified article & associated replies from ID', async () => {
      expect(
        await gql`
          {
            GetArticle(id: "foo") {
              text
              references {
                type
              }
              replyCount
              articleReplies(status: NORMAL) {
                status
                canUpdateStatus
                createdAt
                updatedAt
                articleId
                article {
                  id
                }
                replyId
                reply {
                  id
                  versions {
                    text
                    type
                    reference
                  }
                }
              }
              replyRequestCount
              replyRequests {
                reason
                feedbackCount
                positiveFeedbackCount
                negativeFeedbackCount
              }
              requestedForReply
            }
          }
        `({}, { userId: 'fakeUser', appId: 'LINE' })
      ).toMatchSnapshot();
    });

    it('should allow filtering article replies', async () => {
      expect(
        await gql`
          {
            GetArticle(id: "foo3") {
              articleReplies {
                replyId
                reply {
                  id
                }
                status
              }
              normalReplies: articleReplies(status: NORMAL) {
                replyId
                reply {
                  id
                }
                status
              }
              deletedReplies: articleReplies(status: DELETED) {
                replyId
                reply {
                  id
                }
                status
              }
            }
          }
        `({}, { userId: 'fakeUser', appId: 'LINE' })
      ).toMatchSnapshot();
    });

    it('relatedArticles should work', async () => {
      // No param
      //
      expect(
        await gql`
          {
            GetArticle(id: "foo") {
              relatedArticles {
                edges {
                  cursor
                  node {
                    id
                    text
                  }
                  score
                }
              }
            }
          }
        `()
      ).toMatchSnapshot('relatedArticle no-param test');

      // filter
      //
      expect(
        await gql`
          {
            GetArticle(id: "foo") {
              relatedArticles(filter: { replyCount: { GT: 0 } }) {
                edges {
                  cursor
                  node {
                    id
                    text
                  }
                  score
                }
              }
            }
          }
        `()
      ).toMatchSnapshot('relatedArticle filter test');

      // sort
      //
      expect(
        await gql`
          {
            GetArticle(id: "foo") {
              relatedArticles(orderBy: [{ _score: ASC }]) {
                edges {
                  cursor
                  node {
                    id
                    text
                  }
                  score
                }
              }
            }
          }
        `()
      ).toMatchSnapshot('relatedArticle sorting test');
    });

    it('feedbacks should work', async () => {
      expect(
        await gql`
          {
            GetArticle(id: "foo") {
              articleReplies {
                feedbacks {
                  score
                }
                positiveFeedbackCount
                negativeFeedbackCount
                feedbackCount
              }
            }
          }
        `()
      ).toMatchSnapshot('feedback loading test');
    });
  });

  describe('GetReply', () => {
    it('should get the specified reply & associated articles from ID', async () => {
      expect(
        await gql`
          {
            GetReply(id: "bar") {
              text
              type
              reference
              articleReplies(status: NORMAL) {
                canUpdateStatus
                articleId
                article {
                  text
                }
              }
            }
          }
        `()
      ).toMatchSnapshot();
    });

    it('should allow filtering article replies', async () => {
      expect(
        await gql`
          {
            GetReply(id: "bar2") {
              articleReplies {
                articleId
                article {
                  id
                }
                status
              }
              normalReplies: articleReplies(status: NORMAL) {
                articleId
                article {
                  id
                }
                status
              }
              deletedReplies: articleReplies(status: DELETED) {
                articleId
                article {
                  id
                }
                status
              }
            }
          }
        `()
      ).toMatchSnapshot();
    });
  });

  afterAll(() => unloadFixtures(fixtures));
});
