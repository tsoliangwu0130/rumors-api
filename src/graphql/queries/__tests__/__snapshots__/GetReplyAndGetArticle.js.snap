// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetReplyAndGetArticle GetArticle feedbacks should work: feedback loading test 1`] = `
Object {
  "data": Object {
    "GetArticle": Object {
      "articleReplies": Array [
        Object {
          "feedbackCount": 1,
          "feedbacks": Array [
            Object {
              "score": 1,
            },
          ],
          "negativeFeedbackCount": 0,
          "positiveFeedbackCount": 1,
        },
        Object {
          "feedbackCount": 0,
          "feedbacks": Array [],
          "negativeFeedbackCount": 0,
          "positiveFeedbackCount": 0,
        },
      ],
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetArticle relatedArticles should work: relatedArticle filter test 1`] = `
Object {
  "data": Object {
    "GetArticle": Object {
      "relatedArticles": Object {
        "edges": Array [
          Object {
            "cursor": "WyJkb2MjZm9vMyJd",
            "node": Object {
              "id": "foo3",
              "text": "Lorum ipsum Lorum ipsum Lorum ipsum",
            },
            "score": 0.37905684,
          },
          Object {
            "cursor": "WyJkb2MjZm9vMiJd",
            "node": Object {
              "id": "foo2",
              "text": "Lorum ipsum Lorum ipsum",
            },
            "score": 0.3672113,
          },
        ],
      },
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetArticle relatedArticles should work: relatedArticle no-param test 1`] = `
Object {
  "data": Object {
    "GetArticle": Object {
      "relatedArticles": Object {
        "edges": Array [
          Object {
            "cursor": "WyJkb2MjZm9vMyJd",
            "node": Object {
              "id": "foo3",
              "text": "Lorum ipsum Lorum ipsum Lorum ipsum",
            },
            "score": 0.37905684,
          },
          Object {
            "cursor": "WyJkb2MjZm9vMiJd",
            "node": Object {
              "id": "foo2",
              "text": "Lorum ipsum Lorum ipsum",
            },
            "score": 0.3672113,
          },
        ],
      },
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetArticle relatedArticles should work: relatedArticle sorting test 1`] = `
Object {
  "data": Object {
    "GetArticle": Object {
      "relatedArticles": Object {
        "edges": Array [
          Object {
            "cursor": "WzAuMzY3MjExMywiZG9jI2ZvbzIiXQ==",
            "node": Object {
              "id": "foo2",
              "text": "Lorum ipsum Lorum ipsum",
            },
            "score": 0.3672113,
          },
          Object {
            "cursor": "WzAuMzc5MDU2ODQsImRvYyNmb28zIl0=",
            "node": Object {
              "id": "foo3",
              "text": "Lorum ipsum Lorum ipsum Lorum ipsum",
            },
            "score": 0.37905684,
          },
        ],
      },
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetArticle should allow filtering article replies 1`] = `
Object {
  "data": Object {
    "GetArticle": Object {
      "articleReplies": Array [
        Object {
          "reply": Object {
            "id": "fofo",
          },
          "replyId": "fofo",
          "status": "NORMAL",
        },
        Object {
          "reply": Object {
            "id": "bar2",
          },
          "replyId": "bar2",
          "status": "DELETED",
        },
      ],
      "deletedReplies": Array [
        Object {
          "reply": Object {
            "id": "bar2",
          },
          "replyId": "bar2",
          "status": "DELETED",
        },
      ],
      "normalReplies": Array [
        Object {
          "reply": Object {
            "id": "fofo",
          },
          "replyId": "fofo",
          "status": "NORMAL",
        },
      ],
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetArticle should get the specified article & associated replies from ID 1`] = `
Object {
  "data": Object {
    "GetArticle": Object {
      "articleReplies": Array [
        Object {
          "article": Object {
            "id": "foo",
          },
          "articleId": "foo",
          "canUpdateStatus": false,
          "createdAt": "2015-01-01T12:10:30Z",
          "reply": Object {
            "id": "bar",
            "versions": Array [
              Object {
                "reference": "barbar",
                "text": "bar",
                "type": "NOT_ARTICLE",
              },
            ],
          },
          "replyId": "bar",
          "status": "NORMAL",
          "updatedAt": "2015-01-02T12:10:30Z",
        },
      ],
      "references": Array [
        Object {
          "type": "LINE",
        },
      ],
      "replyCount": 1,
      "replyRequestCount": 1,
      "replyRequests": Array [
        Object {
          "feedbackCount": 2,
          "negativeFeedbackCount": 1,
          "positiveFeedbackCount": 1,
          "reason": "Reason foo",
        },
      ],
      "requestedForReply": true,
      "text": "Lorum ipsum",
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetReply should allow filtering article replies 1`] = `
Object {
  "data": Object {
    "GetReply": Object {
      "articleReplies": Array [
        Object {
          "article": Object {
            "id": "foo2",
          },
          "articleId": "foo2",
          "status": "NORMAL",
        },
        Object {
          "article": Object {
            "id": "foo3",
          },
          "articleId": "foo3",
          "status": "DELETED",
        },
      ],
      "deletedReplies": Array [
        Object {
          "article": Object {
            "id": "foo3",
          },
          "articleId": "foo3",
          "status": "DELETED",
        },
      ],
      "normalReplies": Array [
        Object {
          "article": Object {
            "id": "foo2",
          },
          "articleId": "foo2",
          "status": "NORMAL",
        },
      ],
    },
  },
}
`;

exports[`GetReplyAndGetArticle GetReply should get the specified reply & associated articles from ID 1`] = `
Object {
  "data": Object {
    "GetReply": Object {
      "articleReplies": Array [
        Object {
          "article": Object {
            "text": "Lorum ipsum",
          },
          "articleId": "foo",
          "canUpdateStatus": true,
        },
      ],
      "reference": "barbar",
      "text": "bar",
      "type": "NOT_ARTICLE",
    },
  },
}
`;
